src/server/zset_family.cc

```diff
@@ -2166,64 +2181,106 @@ void ZSetFamily::ZCount(CmdArgList args, const CommandContext& cmd_cntx) {
   }
 }
 
-void ZSetFamily::ZDiff(CmdArgList args, const CommandContext& cmd_cntx) {
-  vector<vector<ScoredMap>> maps(shard_set->size());
+void ZDiffInternal(CmdArgList args, const CommandContext& cmd_cntx, bool store) {
+  vector<OpResult<vector<ScoredMap>>> maps(shard_set->size(), OpStatus::SKIPPED);
+  auto* rb = static_cast<RedisReplyBuilder*>(cmd_cntx.rb);
+  const string_view dest_key = ArgS(args, 0);
```

Comment:
When someone reads the flow of zdiff its not clear why we have the dest_key here
I find the the most common parts of zdiff and zdiffinternal are actually the diff calculation 
but the response is very different and also in the args and fetch there are few differences.
Presonally I would create a function only for the diff calculation which both flows will call and the rest I would keep in the caller flow